Automattic/woocommerce-payments
# Convert MC Settings page to React

Fixes #2947

#### Changes proposed in this Pull Request



This PR 

- Converts the Multi-currency settings page to a single React component 
- Removes the settings saved by `WC_Settings_Page` implementation
- The PR doesn't involve any (important) visual changes, it changes the base that the setting page runs on. 
- Also it contains some code refactoring by moving the related screens in their own folders, 
- adds tests for each component, 
- removes the PHP unit tests for the old settings implementation.

The new folder structure for the components is like this:

```tree
client/
├─ multi-currency/
│  ├─ multi-currency-settings/
│  │  ├─ enabled-currencies-list
│  │  ├─ store-settings
│  ├─ single-currency-settings
```

Multi currency settings component:

![image](https://user-images.githubusercontent.com/3295/134332416-309de6f2-c1c8-40c1-895d-2c22c3927f66.png)

Single currency settings component:

![image](https://user-images.githubusercontent.com/3295/134332712-c65cf065-aad2-4091-9f17-32a991ecadd3.png)








#### Testing instructions





##### Automatic testing

* Run `npm run test:js multi-currency``


##### Manual testing

* Navigate to `WooCommerce > Settings > Multi-Currency` to see the MC settings page.
* Add some currencies, remove some currencies
* Change store settings and save, refresh the page and check if the settings show the correct values
* The workflows should be the same with the previous version, but without page refreshes.
* Also check the MC onboarding testing scenarios on #2786 if something in this PR breaks it.

-------------------


**Post merge**
